corehq/util/es/interface.py

```diff
@@ -1,195 +1,99 @@
+        if doc.get("_id", object()) != doc_id:
+            doc["_id"] = doc_id
+        kw = {} if params is None else params
+        doc_adapter.index(doc, **kw)
 
     def update_doc_fields(self, index_alias, doc_type, doc_id, fields, params=None):
         self._verify_is_alias(index_alias)
-        self.es.update(index_alias, doc_type, doc_id, body={"doc": self._without_id_field(fields)},
-                       params=params or {})
-
-    def _prepare_count_query(self, query):
-        # pagination params are not required and not supported in ES count API
-        query = query.copy()
-        for extra in ['size', 'sort', 'from', 'to', '_source']:
-            query.pop(extra, None)
-        return query
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        kw = {} if params is None else params
+        doc_adapter.update(doc_id, fields, **kw)
 
     def count(self, index_alias, doc_type, query):
-        query = self._prepare_count_query(query)
-        return self.es.count(index=index_alias, doc_type=doc_type, body=query).get('count')
-
-    @staticmethod
-    def _without_id_field(doc):
-        # Field [_id] is a metadata field and cannot be added inside a document.
-        # Use the index API request parameters.
-        return {key: value for key, value in doc.items() if key != '_id'}
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        return doc_adapter.count(query)
 
     def delete_doc(self, index_alias, doc_type, doc_id):
-        self.es.delete(index_alias, doc_type, doc_id)
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        doc_adapter.delete(doc_id)
 
-    def bulk_ops(self, actions, stats_only=False, **kwargs):
-        for action in actions:
-            if '_source' in action:
-                action['_source'] = self._without_id_field(action['_source'])
-        ret = bulk(self.es, actions, stats_only=stats_only, **kwargs)
-        return ret
+    def bulk_ops(self, index_alias, doc_type, actions, **kwargs):
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        return doc_adapter.bulk(actions, **kwargs)
 
-    def search(self, index_alias=None, doc_type=None, body=None, params=None, **kwargs):
+    def search(self, index_alias, doc_type, body=None, params=None, **kwargs):
         self._verify_is_alias(index_alias)
-        results = self.es.search(index=index_alias, doc_type=doc_type, body=body, params=params or {}, **kwargs)
-        self._fix_hits_in_results(results)
-        return results
-
-    def scroll(self, scroll_id=None, body=None, params=None, **kwargs):
-        results = self.es.scroll(scroll_id, body, params=params or {}, **kwargs)
-        self._fix_hits_in_results(results)
-        return results
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        query = {} if body is None else body
+        params = params if params else {}
```

Comment:
nit: consistency with the line above and other places in this file
```suggestion
        params = {} if params is not None else params
```

Reply:
Should that have been `params = {} if params is None else params`?